### ISSUE TITLE
async test rule proposal

### ISSUE THREAD

**Issue by antonk52** [jest-community/eslint-plugin-jest#652]
Hi, I would to propose a new rule which should address the unobvious false passes such as an example below

```js
it('foo', () => {
    expect(Promise.resolve(1)).resolves.toEqual(2)
})
```

This test passes, thought by looking at it should fail. In order to fix it we can either return the promise or make the function async and put await the result/actual value

```js
it('option 1', () => {
    return expect(Promise.resolve(1)).resolves.toEqual(2)
})

it('option 2', async () => {
    await expect(Promise.resolve(1)).resolves.toEqual(2)
    // or
    expect(await Promise.resolve(1)).toEqual(2)
})
```

This looks like we can track such tests with static analyses and also apply autofixes. I would live any feedback on this.

PS: Thank you for the awesome plugin 🙂

**Comment by G-Rath** [comment]
[`jest/valid-expect`](https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/valid-expect.md) already supports checking that promises in tests are properly returned (either via `return` or `await`) :)
